Update theme to work with current steamos

pickledish opened the pull request:
**Update: this description is out of date, the PR no longer changes `image-rendering`, just updates the classes so the theme works again + a bugfix**

Hey! So this one is an interesting story that took a bit of research.

A few weeks ago, I noticed screenshots were a little blurry when viewed on the deck, and I thought, "oh, this is probably due to a bilinear image downscaling, since the screenshot is being displayed smaller than 1280x800".

Then, I found this CSS theme (thanks for making it, by the way!) and assumed it'd solve the problem -- not only is it nicer to view the screenshots in full-screen, but since the screenshot was taken at 1280x800 and then now displayed at that same resolution, the result should be sharp and pixel-perfect, right?

Well -- no! Even with the patch from this repo, screenshots still had the same blurry look, and this confused me.

I tried using the Chrome dev tools to debug the issue, and after some searching, I found the cuprit:

<img width="1162" alt="Screenshot 2024-06-16 at 1 23 04 PM" src="https://github.com/fsworld009/steam-deck-theme-view-media-in-fullscreen/assets/12091397/d702a947-b32c-4199-8afb-a4eed5832108">

The web render's opinion is that the screen is actually 854x534, which is 2/3 of 1280x800 on each axis. Thus, even with a 1280x800 image being displayed fullscreen, it thinks it needs to downscale the image.

(This seems to be due to the system-level UI scaling of the deck not being set at 1 for the internal screen. I found a way to mess with that [here](https://www.reddit.com/r/SteamDeck/comments/14q1hpl/is_there_anything_i_can_do_to_add_the_ui_scale/), and indeed if you use that "secret menu" to set the scaling to 1, and thus have the web renderer's viewport become 1280x800 like it should be, this issue goes away entirely -- but then all the UI elements are small. This isn't the point anyway, I digress)

So, since scaling will apparently always need to happen, we can get rid of this bilinear-blur effect by changing the CSS property `image-rendering` to `pixelated`, which is more like nearest-neighbor and will try its best to keep the image sharp. The end result (while not as perfect as setting the viewport to 1280x800 and needing no scaling at all) is much much improved from the default situation.

So, I thought this'd make a nice improvement to this CSS Loader theme, since you've already done the hard work of finding the right CSS classes and everything. Let me know what you think!

pickledish committed:
Set improved image-rendering for screenshots

pickledish committed:
Bump theme version and update description

fsworld009 commented:
@pickledish Hi, sorry for late response.

Thank you for creating the PR, I tested a bit and here's my finding:

1. If a screenshot is 1280x800 and I view it on Steam deck screen (1280x800), setting `image-rendering: pixelated` does make image look better.
2. However, if I connect the deck to my 1080p monitor, then it actually looks worse when this property is set:
   a. without `pixelated`: 
![Screenshot 2024-06-30 21-06-07](https://github.com/fsworld009/steam-deck-theme-view-media-in-fullscreen/assets/7109839/58224311-85c3-448f-a517-4057d3b1ab3f)
   b. with `pixelated`: ![Screenshot 2024-06-30 21-06-35](https://github.com/fsworld009/steam-deck-theme-view-media-in-fullscreen/assets/7109839/d0f45902-307e-4354-a557-80a1a48d5bf8)
3. Similarly, when I load an old screenshot that I published before (1600x900), it looks worse on Steam Deck's screen when setting `pixelated`

So it seems to me that `pixelated` is only better when the resolution of screenshot and the display is matched. This is quite limited in my opinion, since it's common for original screenshot and current display to have different resolutions. Some examples:

1. Users take screenshots from their PC that hooks to an 2K monitor,  upload them to Steam server, then view it on Steam Deck screen later
2. Users play game on Steam Deck and take screenshot, but later on plugged into a 1080p monitor to view screenshots


If we want to add it, I would say the property needs to be wrapped as an optional toggle. Personally, I feel this is out of the scope of this project, since this project is mostly for making screenshot view fullscreen only.

I would suggest you to create a new Decky CSS project to focus on image quality instead, since it's likely that we will need a more complex configurations, like a [dropdown](https://github.com/suchmememanyskill/Steam-Deck-Theme-Template/blob/main/Sample%20Advanced%20Theme/theme.json#L11) of rendering [properties](https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering), so users can apply whatever option that suite their use cases. This also benefits people that aren't interested in making the view fullscreen.

You are welcome to reuse the CSS selectors from this project.

pickledish committed:
Update shared.css

pickledish committed:
Add pixelated patch

pickledish committed:
Update shared.css

pickledish committed:
Update theme.json

pickledish committed:
Update theme.json

pickledish commented:
Hey man! I have returned from the dead lol

It seems like this plugin actually isn't working anymore (due to the names of the CSS classes being changed)

I spent a minute finding the new classes, and make a fork of this repo that works again:

https://github.com/pickledish/steam-deck-theme-view-media-in-fullscreen/tree/master

If you're interested in pulling that into your repo here, that'd be cool -- I'll wait a bit to hear from you and otherwise attempt to publish the updated version to the decky themes website I suppose 🤔 

> However, if I connect the deck to my 1080p monitor, then it actually looks worse when this property is set:

Also, wanted to say thanks for checking this out! Yeah I tried this as well and you have a good point about it degrading the image quality a lot when the resolutions don't match, thanks for finding that.

> If we want to add it, I would say the property needs to be wrapped as an optional toggle.

Good plan! Implemented it in my fork, definitely better with everyone having the option that fits them best 👍 

> since it's likely that we will need a more complex configurations, like a [dropdown](https://github.com/suchmememanyskill/Steam-Deck-Theme-Template/blob/main/Sample%20Advanced%20Theme/theme.json#L11) of rendering [properties](https://developer.mozilla.org/en-US/docs/Web/CSS/image-rendering),

(I was curious about this as well, but actually it seems like there are only 2 -- `smooth` (which is bilinear) and `crisp-edges` and `pixelated` (which are the same, both are nearest neighbor). The rest are just that usual jumble of `inherit` and `auto` and all the other CSS stuff. So I think we're OK with a checkbox?

> This also benefits people that aren't interested in making the view fullscreen.

And actually -- I think the `image-rendering: pixelated` patch only makes sense in the context of the fullscreen-media that this project enables! Otherwise, it's that same problem we were talking about -- the resolutions of the screen and the image don't line up and `pixelated` looks bad, right

pickledish committed:
Merge branch 'patch-1' into master

pickledish committed:
Update shared.css

pickledish committed:
Update theme.json

pickledish committed:
Merge pull request #1 from pickledish/master

Merge

pickledish commented on View Media in Fullscreen/shared.css:
```diff
@@ -1,18 +1,12 @@
   z-index: 10000;
 }
 
-/* Patch images with Spoiler alert */
-.mediapage_FullscreenScreenshot_35IHD .mediapage_Screenshot_3Qh8h .mediapage_ImageContainer_120HU .mediapage_Image_dmMTv.mediapage_Spoiler_QIvhs {
-  position: fixed;
-}
-
```
(just got rid of this (and `immediately_apply_theme.css`) since IIUC their CSS selectors are out of date and I don't know what they did anyway, seems to work OK without em)

pickledish commented:
Anyway -- updated this PR accordingly, in case you like it now 👍

pickledish commented:
Preview:

![IMG_0611](https://github.com/user-attachments/assets/0bd5c917-30dc-4e1f-84d1-006b5cc6a7e3)

pickledish committed:
99.9vw removes need for patch

pickledish commented:
Huh -- was playing around a bit and setting `99.9vw` instead of `100vw` gets the image to look sharp on the deck screen without needing to set `image-rendering: pixelated` at all. Weird. So that should be the best of both worlds I guess, updated this PR and removed the patch file
